src/main/kotlin/racingcar/domain/RacingGame.kt

```diff
@@ -0,0 +1,17 @@
+package racingcar.domain
+
+class RacingGame(
+    private val cars: List<Car>,
+    private val tryCount: Int,
+) {
+    fun run(engine: Engine): List<RacingGameResult> {
+        return List(tryCount) { runGameOneStage(engine) }
+    }
+
+    private fun runGameOneStage(engine: Engine): RacingGameResult {
+        cars.forEach { car ->
+            car.moveOrStop(engine)
+        }
+        return RacingGameResult(cars.map { it.copy() })
```

Comment:
처음부터 moveOrStop 의 결과가 (깊은)복사를 한 Car 객체를 반환한다면 아래와 같이 변경할 수 있을 것 같아요. 😉

```suggestion
        return cars.map { it.moveOrStop(engine) }
```